arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/stream/Stream.kt

```diff
@@ -602,6 +603,71 @@ inline fun <A> StreamOf<A>.fix(): Stream<A> =
   fun <B> effectMap(f: suspend (O) -> B): Stream<B> =
     flatMap { o -> effect { f(o) } }
 
+  /**
+   * Like [effectMap], but will evaluate effects in parallel, emitting the results
+   * downstream in the same order as the input stream. The number of concurrent effects
+   * is limited by [maxConcurrent].
+   *
+   * See [parEffectMapUnordered] if there is no requirement to retain the order of the original stream.
+   *
+   * ```kotlin:ank:playground
+   * import arrow.fx.coroutines.stream.*
+   *
+   * //sampleStart
+   * suspend fun main(): Unit =
+   *   Stream(1, 2, 3, 4)
+   *     .parEffectMap(2) { print(it) }
+   *     .compile()
+   *     .drain() // 1234
+   * //sampleEnd
+   * ```
+   */
+  fun <O2> parEffectMap(maxConcurrent: Int, f: suspend (O) -> O2): Stream<O2> =
+    effect {
+      Pair(Queue.unbounded<(suspend () -> Either<Throwable, O2>)?>(), Promise<Unit>())
```

Comment:
No, since here we're building a queue of suspending operations that we can then `pull`.

So we drain our stream, and in order run every `f` in a new `suspend` function which is executed when the resulting stream pulls. We need to do this to ensure that the order is maintained when pulling.

Reply:
Shouldn't ordering should be the same among those 2, only differentiating on the amount of elements they can contain? Also this (bounded) is what the original impl is doing